src/psyclone/tests/psyad/domain/lfric/test_lfric_adjoint.py

```diff
@@ -57,81 +68,543 @@ def test_generate_lfric_adjoint_no_container_error(fortran_reader):
+    metadata.validate()
+    # This can be used for any additional arguments as we don't use
+    # these in our tests.
+    dummy = DataSymbol("dummy", INTEGER_TYPE,
+                       interface=ArgumentInterface(
+                           ArgumentInterface.Access.READ))
+    # It doesn't matter that field_1, field_2, field_3 and operator
+    # are the wrong type as we only care about their intent.
+    field_1 = DataSymbol("field_1", REAL_TYPE,
+                         interface=ArgumentInterface(
+                             ArgumentInterface.Access.READWRITE))
+    field_2 = DataSymbol("field_2", REAL_TYPE,
+                         interface=ArgumentInterface(
+                             ArgumentInterface.Access.READWRITE))
+    field_3 = DataSymbol("field_3", REAL_TYPE,
+                         interface=ArgumentInterface(
+                             ArgumentInterface.Access.READWRITE))
+    scalar = DataSymbol("scalar", REAL_TYPE,
+                        interface=ArgumentInterface(
+                            ArgumentInterface.Access.READWRITE))
+    operator = DataSymbol("operator", REAL_TYPE,
+                          interface=ArgumentInterface(
+                              ArgumentInterface.Access.READWRITE))
+    return (metadata, dummy, field_1, field_2, field_3, scalar, operator)
+
+
+def test_update_access_metadata_none():
+    '''Test that the update_access_metadata method returns None and does
+    not modify the metadata if the supplied symbol name is not an
+    argument.
+
+    '''
+    metadata, dummy, field_1, field_2, field_3, scalar, operator = \
+        get_metadata_args()
+    assert metadata.meta_args[0].access.lower() == "gh_write"
+    assert metadata.meta_args[1].access.lower() == "gh_read"
+    assert metadata.meta_args[2].access.lower() == "gh_read"
+    assert metadata.meta_args[3].access.lower() == "gh_read"
+    assert metadata.meta_args[4].access.lower() == "gh_read"
+    arguments = [
+        dummy, dummy, field_1, field_2, field_3, scalar, dummy, operator]
+    access = _update_access_metadata("internal", arguments, metadata)
+    assert access is None
+    assert metadata.meta_args[0].access.lower() == "gh_write"
+    assert metadata.meta_args[1].access.lower() == "gh_read"
+    assert metadata.meta_args[2].access.lower() == "gh_read"
+    assert metadata.meta_args[3].access.lower() == "gh_read"
+    assert metadata.meta_args[4].access.lower() == "gh_read"
+
+
+def test_update_access_metadata_index_error():
+    '''Test that the _update_access_metadata method raises the expected
+    exception if the position of the active variable does not match
+    the position expected by the metadata.
+
+    '''
+    metadata, _1, field_1, _2, _3, _4, _6 = get_metadata_args()
+    arguments = [field_1]
+    with pytest.raises(GenerationError) as info:
+        _ = _update_access_metadata("field_1", arguments, metadata)
+    assert ("The argument position '0' of the active variable 'field_1' "
```

Comment:
Could you include the kernel name too if possible?

Reply:
No the way it is written as the method `_update_access_metadata` is not given this information. I could re-raise outside of `_update_access_metadata` and add the kernel name, although PSyAD only works on one kernel at a time anyway.